core/app/models/spree/tax/item_adjuster.rb

```diff
@@ -0,0 +1,32 @@
+module Spree
+  module Tax
+    class ItemAdjuster
+      attr_reader :item, :order
+
+      include TaxHelpers
+
+      def initialize(item, options = {})
+        @item = item
+        @order = @item.order
+        # set the instance variable so `TaxRate.match` is only called when necessary
+        @rates_for_order_zone = options[:rates_for_order_zone]
+      end
+
+      def adjust!
```

Comment:
I can very well imagine a future non-bang `adjust` method: One that will not change the order, and not create adjustments, but just `build` them. That would be great to reduce database calls when updating items and orders. Alas, we're not there yet, but I'd like to keep this method's name the same at that point. 

YARD I can absolutely add, as this method returns the newly created adjustments (making it somewhat easier to test than `Spree::TaxRate.adjust`).